src/libraries/System.Private.CoreLib/src/System/WeakReference.T.cs

```diff
@@ -63,5 +64,67 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
             info.AddValue("TrackedObject", this.Target, typeof(T)); // Do not rename (binary serialization)
             info.AddValue("TrackResurrection", IsTrackResurrection()); // Do not rename (binary serialization)
         }
+
+#if !CORECLR
+        // Most methods using the handle should use GC.KeepAlive(this) to avoid potential handle recycling
+        // attacks (i.e. if the WeakReference instance is finalized away underneath you when you're still
+        // handling a cached value of the handle then the handle could be freed and reused).
+
+        // the handle field is effectively readonly, not formally readonly because we assign it in "Create".
+        // the lowermost bit is used to indicate whether the handle is tracking resurrection
+        private IntPtr _handleAndKind;
```

Comment:
Having the data fields in the middle of the class definition looks odd. The coding style is to have them at the top of the class definition. It may look better to have them at the top.